src/specialty.jl

```diff
@@ -17,26 +17,486 @@ end
+## Separable kernel
+
+A kernel is called *separable* if it can be expressed as the
+convolution of two one-dimensional filters. With a matrix representation
+of the kernel, separability means that the kernel matrix can be written
+as an outer product of two vectors. Separable kernels offer
+computational advantages since instead of performing a two-dimensional
+convolution one can perform a sequence of one-dimensional convolutions.
+
+# Options
+You can specify your choice of the finite-difference scheme via the `kernelfun`
+parameter. You can also indicate how to deal with the pixels on the border
+of the image with the `border` parameter.
+
+## Choices for `kernelfun`
+In general `kernelfun` can be any function which satisfies the following
+interface:
+```julia
+    kernelfun(extended::NTuple{N,Bool}, d) -> kern_d,
+```
+where `kern_d` is the kernel for producing the derivative with respect to
+the ``d``th dimension of an ``N``-dimensional array. The parameter `extended[i]` is true
+if the image is of size > 1 along dimension ``i``. The parameter `kern_d` may be
 provided as a dense or factored kernel, with factored representations
 recommended when the kernel is separable.
+
+Some valid `kernelfun` options are described below.
+
+### `KernelFactors.prewitt`
+
+With the *prewit* option the computation of the gradient is based on
+the kernels
+```math
+\\begin{aligned}
+\\mathbf{H}_{x_1} & = \\frac{1}{6}
+    \\begin{bmatrix}
+    -1 & -1 & -1 \\\\
+    0 & 0 & 0 \\\\
+    1 & 1 & 1
+    \\end{bmatrix}
+&
+\\mathbf{H}_{x_2} & =  \\frac{1}{6}
+    \\begin{bmatrix}
+    -1 & 0 & 1 \\\\
+    -1 & 0 & 1 \\\\
+    -1 & 0 & 1
+    \\end{bmatrix} \\\\
+& = \\frac{1}{6}
+    \\begin{bmatrix}
+    1 \\\\
+    1  \\\\
+    1
+    \\end{bmatrix}
+    \\begin{bmatrix}
+    -1 & 0 & 1
+    \\end{bmatrix}
+&
+& = \\frac{1}{6}
+    \\begin{bmatrix}
+    -1 \\\\
+    0  \\\\
+    1
+    \\end{bmatrix}
+    \\begin{bmatrix}
+    1 & 1 & 1
+    \\end{bmatrix}.
+\\end{aligned}
+```
+See also: [`prewitt`](@ref), [`KernelFactors.prewitt`](@ref) and
+[`Kernel.prewitt`](@ref)
+
+### `KernelFactors.sobel`
+
+The *sobel* option designates the kernels
+```math
+\\begin{aligned}
+\\mathbf{H}_{x_1} & = \\frac{1}{8}
+    \\begin{bmatrix}
+    -1 & -2 & -1 \\\\
+     0 & 0 & 0 \\\\
+     1 & 2 & 1
+    \\end{bmatrix}
+&
+\\mathbf{H}_{x_2} & = \\frac{1}{8}
+    \\begin{bmatrix}
+    -1 & 0 & 1 \\\\
+    -2 & 0 & 2 \\\\
+    -1 & 0 & 1
+    \\end{bmatrix} \\\\
+& = \\frac{1}{8}
+    \\begin{bmatrix}
+    -1 \\\\
+    0  \\\\
+    1
+    \\end{bmatrix}
+    \\begin{bmatrix}
+    1 & 2 & 1
+    \\end{bmatrix}
+&
+& = \\frac{1}{8}
+    \\begin{bmatrix}
+    1 \\\\
+    2  \\\\
+    1
+    \\end{bmatrix}
+    \\begin{bmatrix}
+    -1 & 0 & 1
+    \\end{bmatrix}.
+\\end{aligned}
+```
+See also: [`sobel`](@ref), [`KernelFactors.sobel`](@ref) and
+[`Kernel.sobel`](@ref)
+
+### `KernelFactors.ando3`
+The *ando3* option specifies the kernels
```

Comment:
For consistency with *scharr* and *bickley*, 
```
The *ando3* option [6] specifies the kernels
```
(although you might want to renumber the references)

Reply:
Done